### PR TITLE
Adding the golangci-lint linter to the repository (jenkinsfile)

### PR DESCRIPTION
**Pull request by jlvoiseux**
Resolves https://github.com/elastic/apm-aws-lambda/issues/153. The chosen linter is [golangci-lint](https://golangci-lint.run/). For this third iteration, we install and run the linter as part of the Lint stage in the Jenkinsfile.

### CODE REVIEW

**Comment by jlvoiseux** [comment]
Update on this PR : the linter was successfully implemented by commit [8907a3a](https://github.com/elastic/apm-aws-lambda/pull/157/commits/8907a3a836d65e2a092245870a4166e9c431314c).

**Failing Linter output**
```
11:54:39  + ./golangci-lint.sh
11:54:39  golangci/golangci-lint info checking GitHub for tag 'v1.45.0'
11:54:39  golangci/golangci-lint info found version: 1.45.0 for v1.45.0/linux/amd64
11:54:40  golangci/golangci-lint info installed /var/lib/jenkins/workspace/ibrary_apm-aws-lambda-mbp_PR-157/bin/golangci-lint
11:54:41  golangci-lint has version 1.45.0 built from 1f4c1ed9 on 2022-03-18T15:08:39Z
11:54:50  main.go:70:27: Error return value of `extension.StartHttpServer` is not checked (errcheck)
11:54:50  	extension.StartHttpServer(agentDataChannel, config)
11:54:50  	                         ^
11:54:50  main_test.go:168:12: Error return value of `client.Do` is not checked (errcheck)
11:54:50  		client.Do(reqData)
11:54:50  		         ^
11:54:50  main_test.go:189:14: Error return value of `client.Do` is not checked (errcheck)
11:54:50  				client.Do(reqData)
11:54:50  				         ^
11:54:50  e2e-testing/e2e_test.go:135:9: Error return value of `io.Copy` is not checked (errcheck)
11:54:50  	io.Copy(out, resp.Body)
11:54:50  	       ^
11:54:50  e2e-testing/e2e_test.go:150:11: Error return value of `os.Chmod` is not checked (errcheck)
11:54:50  		os.Chmod(filepath.Join(agentFolderPath, f.Name()), 0755)
11:54:50  		        ^
11:54:50  e2e-testing/e2e_util.go:38:9: Error return value of `e.Start` is not checked (errcheck)
11:54:50  	e.Start()
11:54:50  	       ^
11:54:50  e2e-testing/e2e_util.go:49:8: Error return value of `e.Wait` is not checked (errcheck)
11:54:50  	e.Wait()
11:54:50  	      ^
11:54:50  e2e-testing/e2e_util.go:78:13: Error return value of `os.MkdirAll` is not checked (errcheck)
11:54:50  	os.MkdirAll(destinationFolderPath, 0755)
11:54:50  	           ^
11:54:50  e2e-testing/e2e_util.go:100:15: Error return value of `os.MkdirAll` is not checked (errcheck)
11:54:50  			os.MkdirAll(path, f.Mode())
11:54:50  			           ^
11:54:50  e2e-testing/e2e_util.go:102:15: Error return value of `os.MkdirAll` is not checked (errcheck)
11:54:50  			os.MkdirAll(filepath.Dir(path), f.Mode())
11:54:50  			           ^
11:54:50  logsapi/subscribe.go:99:15: Error return value of `Server.Serve` is not checked (errcheck)
11:54:50  		Server.Serve(Listener)
11:54:50  		            ^
11:54:50  extension/apm_server_test.go:38:11: Error return value of `gw.Write` is not checked (errcheck)
11:54:50  		gw.Write([]byte(s))
11:54:50  		        ^
11:54:50  extension/apm_server_test.go:52:10: Error return value of `w.Write` is not checked (errcheck)
11:54:50  		w.Write([]byte(`{"foo": "bar"}`))
11:54:50  		       ^
11:54:50  extension/apm_server_test.go:74:11: Error return value of `gw.Write` is not checked (errcheck)
11:54:50  		gw.Write(body)
11:54:50  		        ^
11:54:50  extension/apm_server_test.go:85:10: Error return value of `w.Write` is not checked (errcheck)
11:54:50  		w.Write([]byte(`{"foo": "bar"}`))
11:54:50  		       ^
11:54:50  extension/apm_server_test.go:178:11: Error return value of `gw.Write` is not checked (errcheck)
11:54:50  		gw.Write([]byte(""))
11:54:50  		        ^
11:54:50  extension/apm_server_test.go:192:10: Error return value of `w.Write` is not checked (errcheck)
11:54:50  		w.Write([]byte(`{"foo": "bar"}`))
11:54:50  		       ^
11:54:50  extension/apm_server_test.go:201:17: Error return value is not checked (errcheck)
11:54:50  	PostToApmServer(apmServer.Client(), agentData, &config, context.Background())
11:54:50  	               ^
11:54:50  extension/apm_server_test.go:244:17: Error return value is not checked (errcheck)
11:54:50  	PostToApmServer(apmServer.Client(), agentData, &config, context.Background())
11:54:50  	               ^
11:54:50  extension/apm_server_test.go:285:17: Error return value is not checked (errcheck)
11:54:50  	PostToApmServer(apmServer.Client(), agentData, &config, context.Background())
11:54:50  	               ^
11:54:50  extension/apm_server_test.go:346:10: Error return value of `io.Copy` is not checked (errcheck)
11:54:50  		io.Copy(ioutil.Discard, r.Body)
11:54:50  		       ^
11:54:50  extension/http_server.go:48:24: Error return value of `agentDataServer.Serve` is not checked (errcheck)
11:54:50  		agentDataServer.Serve(ln)
11:54:50  		                     ^
11:54:50  e2e-testing/e2e_util.go:56:2: S1008: should use 'return err == nil' instead of 'if err == nil { return true }; return false' (gosimple)
11:54:50  	if err == nil {
11:54:50  	^
11:54:50  e2e-testing/e2e_test.go:46:16: S1028: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (gosimple)
11:54:50  		ProcessError(errors.New(fmt.Sprintf("Unsupported language %s ! Supported languages are %v", languageName, supportedLanguages)))
11:54:50  		             ^
11:54:50  main_test.go:234:26: S1030: should use bufRecord.String() instead of string(bufRecord.Bytes()) (gosimple)
11:54:50  	logEvent.StringRecord = string(bufRecord.Bytes())
11:54:50  	                        ^
```

Corrections required for the linter to pass were done in ad51c2a.

**Successful Linter output**

```
14:56:58  + ./golangci-lint.sh
14:56:58  golangci/golangci-lint info checking GitHub for tag 'v1.45.0'
14:56:58  golangci/golangci-lint info found version: 1.45.0 for v1.45.0/linux/amd64
14:56:59  golangci/golangci-lint info installed /var/lib/jenkins/workspace/ibrary_apm-aws-lambda-mbp_PR-157/bin/golangci-lint
14:57:00  golangci-lint has version 1.45.0 built from 1f4c1ed9 on 2022-03-18T15:08:39Z
```

To run the linter on your own machine, just run the `./golangci-lint.sh` script.

File: apm-lambda-extension/extension/apm_server_test.go [hunk 1]
```diff
@@ -274,16 +352,18 @@ func TestAPMServerRecovery(t *testing.T) {
 		bytes, _ := ioutil.ReadAll(r.Body)
 		assert.Equal(t, string(data), string(bytes))
 		assert.Equal(t, "gzip", r.Header.Get("Content-Encoding"))
-		w.Write([]byte(`{"foo": "bar"}`))
+		_, err := w.Write([]byte(`{"foo": "bar"}`))
```
Comment (estolfo) [inline apm-lambda-extension/extension/apm_server_test.go:355]: You could use this syntax in a few places in this PR, if you want:
```go
if _, err := w.Write([]byte(`{"foo": "bar"}`)); err != nil {
  return
}
```

File: apm-lambda-extension/extension/apm_server_test.go [hunk 2]
```diff
@@ -302,9 +382,21 @@ func TestContinuedAPMServerFailure(t *testing.T) {
 	pr, pw := io.Pipe()
 	gw, _ := gzip.NewWriterLevel(pw, gzip.BestSpeed)
 	go func() {
-		gw.Write([]byte(""))
-		gw.Close()
-		pw.Close()
+		_, err := gw.Write([]byte(""))
```
Comment (estolfo) [inline apm-lambda-extension/extension/apm_server_test.go:385]: for example,

```go
if _, err := gw.Write([]byte("")); err != nil {
  t.Fail()
  return
}
```

**Review by estolfo — COMMENTED** [review]
I just had a comment about using the shorthand `; err != nil {`
That's not a necessary change, just up to you if you want to shave off some more lines of code.

File: apm-lambda-extension/extension/apm_server_test.go (hunk 2 above)
Comment (jlvoiseux) [reply apm-lambda-extension/extension/apm_server_test.go]: Thanks for the advice ! In a34cd53 and 1bdfc14, I have used the shorthand wherever I could use it.

File: apm-lambda-extension/e2e-testing/e2e_test.go [hunk 3]
```diff
@@ -40,10 +39,12 @@ func TestEndToEnd(t *testing.T) {
 		t.Skip("Skipping E2E tests. Please set the env. variable RUN_E2E_TESTS=true if you want to run them.")
 	}
 
+	extension.Log.Info("If the end-to-end tests are failing unexpectedly, please verify that Docker is running on your machine.")
+
```
[the remark this answers is not in the document]
Comment (jlvoiseux) [reply apm-lambda-extension/e2e-testing/e2e_test.go:43]: Exactly :D

**Review by estolfo — APPROVED** [review]

**Pull request merged**